### PR TITLE
Implement MatchData

### PR DESCRIPTION
**Pull request by saveriomiroddi**
The `MatchData` class allows captures matching with regular expressions:

    md = Regexp.new("a (\w+)" ).match("This is a flower") #=> #<MatchData "a flower" 1:"flower">
    md.captures[0] #=> "flower"

### CODE REVIEW

**Comment by hachi8833** [comment]
Amazing!
I think you can move `match` to Regexp. How about this?

**Comment by saveriomiroddi** [comment]
> I think you can move match to Regexp. How about this?

@hachi8833 sorry, can you point it in the code (eg. via review)? if you refer to the method, it's already defined as `Regexp#match`.

@st0012 the comment above actually made me think about where regex matching methods should be located. in ruby, they're all located in `String`, so that when one wants to perform a match, the string is always the left element, and the regex the right one:

    string (operation) regex

right now, methods are begin added to Regexp (even this PR), so that there is an inconsistency between the invocation patterns:

    string =~ regex
    regexp.match?(string)

is this intended? do you have any design direction?

**Comment by st0012** [comment]
@saveriomiroddi Putting the method in String class seems to be a better choice.

**Comment by saveriomiroddi** [comment]
PR completed.

File: vm/match_data_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	"testing"
+)
+
+func TestMatchDataClassSuperclass(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{`String.class.name`, "Class"},
```
Comment (st0012) [inline vm/match_data_test.go]: I think this should be `MatchData`?

Comment (saveriomiroddi) [reply vm/match_data_test.go]: Nuts! Fixed :smile:

**Pull request merged**